### PR TITLE
[DBX-88811] treat LH BI success state as true success

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- We will now treat a Benefits Intake status of `success` as true success in our system
- follow up work to add 'paraniod' background polling to watch for submissions transitioning out of `success`
- This unblocks "definition of success" work that is foundational to 526 state monitoring
- DBX team carbs

## Related issue(s)

- [Ticket](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/88811)
- [Slack thread where we settle on this with OCTO and LH](https://dsva.slack.com/archives/C053U7BUT27/p1721405039490299?thread_ts=1721337336.243199&cid=C053U7BUT27)
- [Document on 526 state and reporting](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/untouched_submission_audit/526_state_repair_tdd.md)

## Testing done

- [x] *New code is covered by unit tests*
- Manually tested in production command line as part of remediation backfil

## What areas of the site does it impact?
form 526 state

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (comming soon)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature (n/a)

### CODE REVIEW

File: spec/models/form526_submission_spec.rb [hunk 1]
```diff
@@ -108,6 +116,28 @@
...
+    describe 'paranoid_success_type' do
+      it 'returns records less than a year old with paranoid_success backup status' do
+        expect(Form526Submission.paranoid_success_type).to contain_exactly(
+          paranoid_success
+        )
+      end
+    end
+
+    describe 'success_by_age_type' do
+      it 'returns records less than a year old with paranoid_success backup status' do
```
Comment (freeheeling) [inline spec/models/form526_submission_spec.rb:135]: Should this be 'records **more** than a year old'?

Comment (SamStuckey) [reply spec/models/form526_submission_spec.rb:135]: yessir immaculate catch

File: spec/models/form526_submission_spec.rb [hunk 2]
```diff
@@ -168,13 +198,15 @@
...
+          accepted_backup,
+          paranoid_success,
+          success_by_age
         )
       end
     end
 
     describe 'failure_type' do
-      it 'returns anything not explicitly successful or still in process' do
+      it 'returns anything not successful of in process' do
```
Comment (freeheeling) [inline spec/models/form526_submission_spec.rb:209]: Should this be '**or** in process'?

File: app/sidekiq/form526_status_polling_job.rb [hunk 3]
```diff
@@ -34,33 +37,45 @@ def api_to_poll
...
+  def handle_submission(status, form_submission)
+    if %w[error expired].include? status
+      log_result('failure')
+      form_submission.rejected!
+    elsif status == 'vbms'
+      log_result('true success')
+      form_submission.accepted!
+    elsif status == 'success' && !paranoid
+      log_result('paranoid success')
+      form_submission.paranoid_success!
```
Comment (freeheeling) [inline app/sidekiq/form526_status_polling_job.rb:66]: I'm confused here. Isn't 'paranoid success' an indication of 'success' and `paranoid` is **true**, not if `paranoid` is **false**? Maybe I'm misinterpreting the evaluation.

File: spec/models/form526_submission_spec.rb (hunk 2 above)
Comment (SamStuckey) [reply spec/models/form526_submission_spec.rb:209]: just reverting it to the original actually

File: app/sidekiq/form526_status_polling_job.rb (hunk 3 above)
Comment (SamStuckey) [reply app/sidekiq/form526_status_polling_job.rb:66]: yeah it's weird... here's my thought process

When we run `paranoid: true` we are scoping our poll to submissions that are already set to `paranoid_success`. Therefor, if we are at this point during a `paranoid: true` run, and `status == 'success'`, then there is no reason to call `paranoid_success!`, simply because it would be redundant. 

However if we are running `paranoid: false` and return `status == 'success'`, then we are polling on a submission that currently is pending, but now should be marked paranoid_success. therefor, `status == 'success'` and `paranoid: false` indicates we should mark this a paranoid success.

Comment (SamStuckey) [reply app/sidekiq/form526_status_polling_job.rb]: added

Comment (freeheeling) [reply app/sidekiq/form526_status_polling_job.rb]: That makes sense. Thanks for the explanation.

**Review by freeheeling — DISMISSED** [review]
nice!

**Review by freeheeling — APPROVED** [review]
✅

**Comment by rmtolmach** [comment]
> Manually tested in production command line as part of remediation backfil
Anything involving "production command line" make me a little nervous 😅. Could it not have been done in staging or dev? What were the testing steps?

**Comment by SamStuckey** [comment]
@rmtolmach Ideally that would be the case. However, this code is part of Code Yellow remediation effort that are happening anyway, via scripts in the production command line, so it's simpler to just test this against the same submissions in production. The 'saftey' here is that this doesn't effect anything other than fields no one but me is using at the moment, ie. taggs on remediated submissions. i can set and unset them with confidence and there are redundancies in my process to ensure they are set correctly at final runtime.

TL;DR - yes it's gross, but this whole code yellow has been like this. Soon the Code Yellow server we are using will be killed and we won't have to do this any more

**Comment by SamStuckey** [comment]
I should clarify how this job connects to the scripts i mentioned. I'm running these scripts daily in production as part of Code Yellow remediation. This job is a way to not have to do that anymore. Ergo this code has been well vetted in production, because we had no other choice, and now by committing it to the codebase we can stop using a production command line

**Review by rmtolmach — APPROVED** [review]
Thanks for the explanation! ☝️ That helps. 🚀
